diffpy/snmf/subroutines.py

```diff
@@ -28,38 +29,38 @@ def lift_data(data_input, lift=1):
     return data_input + np.abs(np.min(data_input) * lift)
 
 
-def initialize_arrays(number_of_components, number_of_moments, signal_length):
-    """Generates the initial guesses for the weight, stretching, and component matrices
-
-    Calculates the initial guesses for the component matrix, stretching factor matrix, and weight matrix. The initial
-    guess for the component matrix is a random (signal_length) x (number_of_components) matrix where each element is
-    between 0 and 1. The initial stretching factor matrix is a random (number_of_components) x (number_of_moments)
-    matrix where each element is number slightly perturbed from 1. The initial weight matrix guess is a random
-    (number_of_components) x (number_of_moments) matrix where each element is between 0 and 1.
+def create_components(number_of_components, grid_vector, number_of_signals, signal_length, perturbation=1e-3):
+    """Creates the ComponentSignal objects
 
     Parameters
     ----------
     number_of_components: int
-      The number of component signals to obtain from the stretched nmf decomposition.
-
-    number_of_moments: int
-      The number of signals in the user provided dataset where each signal is at a different moment.
-
+      the number of component signals in the NMF decomposition
+    grid_vector: 1d array
+      The 1d array containing the grid of the signals provided by the user.
+    number_of_signals: int
+      The number of signals in the data input provided by the user.
     signal_length: int
-      The length of each signal in the user provided dataset.
+      The number specifying the length of the signals' grid.
+    perturbation: float
+      The number used in creating the stretching factor guess. Specifies how much the elements should be perturbed
+      from one.
+
 
     Returns
     -------
-    tuple of 2d arrays of floats
-      The tuple containing three elements: the initial component matrix guess, the initial stretching factor matrix
-      guess, and the initial weight factor matrix guess in that order.
+    tuple of ComponentSignal objects
+      The tuple containing a series of ComponentSignals
```

Comment:
This is confusing to me....what does "series" mean? maybe `The initialized but empty ComponentSignal objects, one per component.'